From eda13358f921c80dc3c075f36860a82bfb151bef Mon Sep 17 00:00:00 2001
From: Pavel Zhukov <pzhukov@redhat.com>
Date: Thu, 21 Feb 2019 10:24:24 +0100
Subject: [PATCH] Change default requested options

Add NIS domain, NIS servers, NTP servers, interface-mtu and domain-search
to the list of default requested DHCP options

---
 client/clparse.c | 27 ++++++++++++++++++++++++++-
 1 file changed, 26 insertions(+), 1 deletion(-)

diff --git a/client/clparse.c b/client/clparse.c
index 7212e3a..39b95a0 100644
--- a/client/clparse.c
+++ b/client/clparse.c
@@ -31,7 +31,7 @@
 
 struct client_config top_level_config;
 
-#define NUM_DEFAULT_REQUESTED_OPTS	9
+#define NUM_DEFAULT_REQUESTED_OPTS	14
 /* There can be 2 extra requested options for DHCPv4-over-DHCPv6. */
 struct option *default_requested_options[NUM_DEFAULT_REQUESTED_OPTS + 2 + 1];
 
@@ -116,6 +116,31 @@ isc_result_t read_client_conf ()
 	option_code_hash_lookup(&default_requested_options[8],
 				dhcpv6_universe.code_hash, &code, 0, MDL);
 
+	/* 10 */
+	code = DHO_NIS_DOMAIN;
+	option_code_hash_lookup(&default_requested_options[9],
+				dhcp_universe.code_hash, &code, 0, MDL);
+
+	/* 11 */
+	code = DHO_NIS_SERVERS;
+	option_code_hash_lookup(&default_requested_options[10],
+				dhcp_universe.code_hash, &code, 0, MDL);
+
+	/* 12 */
+	code = DHO_NTP_SERVERS;
+	option_code_hash_lookup(&default_requested_options[11],
+				dhcp_universe.code_hash, &code, 0, MDL);
+
+	/* 13 */
+	code = DHO_INTERFACE_MTU;
+	option_code_hash_lookup(&default_requested_options[12],
+				dhcp_universe.code_hash, &code, 0, MDL);
+
+	/* 14 */
+	code = DHO_DOMAIN_SEARCH;
+	option_code_hash_lookup(&default_requested_options[13],
+				dhcp_universe.code_hash, &code, 0, MDL);
+
 	for (code = 0 ; code < NUM_DEFAULT_REQUESTED_OPTS ; code++) {
 		if (default_requested_options[code] == NULL)
 			log_fatal("Unable to find option definition for "
